File: IT/Stringhe/Esercizi/Python/esercizio25.py
# file esercizio25.py

from __future__ import print_function, unicode_literals
import re
if hasattr(__builtins__, 'raw_input'): input = raw_input

cod = ""
pattern = re.compile(r"[A-Z]{6}[\d]{2}[ABCDEHLMPRST]{1}(?:0[1-9]|[1-6][\d]|7[01])[A-Z0-9]{4}[A-Z]{1}$")

while not cod:
	s = input("Digita il tuo codice fiscale: ")
	if pattern.match(s):
		cod = s
		print("Codice fiscale corretto.")
	else:
		print("Codice fiscale non valido!")

File: IT/Stringhe/Esercizi/Java/Esercizio25.java
/* file Esercizio25.java */

import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Esercizio25 {
	public static void main(String [] args) {
		String cod = "";

		String r = "^[A-Z]{6}[\\d]{2}[ABCDEHLMPRST]{1}(?:0[1-9]|[1-6][\\d]|7[01])[A-Z0-9]{4}[A-Z]{1}$";
		Pattern pattern = Pattern.compile(r);

		while (cod.isEmpty()) {
			String s = System.console().readLine("Digita il tuo codice fiscale: ");
			if (pattern.matcher(s).matches()) {
				cod = s;
				System.out.println("Codice fiscale corretto.");
			}
			else {
				System.out.println("Codice fiscale non valido!");
			}
		}
	}
}

